// semantic-ui-vue/dist/commonjs/modules/Popup/Popup.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _VFragment = require('../../lib/VFragment');

var _VFragment2 = _interopRequireDefault(_VFragment);

var _lib = require('../../lib');

var _PropTypes = require('../../lib/PropTypes');

var _PopupContainer = require('./private/PopupContainer');

var _PopupContainer2 = _interopRequireDefault(_PopupContainer);

var _popupConstants = require('./private/popupConstants');

var _PopupHeader = require('./PopupHeader');

var _PopupHeader2 = _interopRequireDefault(_PopupHeader);

var _PopupContent = require('./PopupContent');

var _PopupContent2 = _interopRequireDefault(_PopupContent);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = {
  name: 'SuiPopup',
  description: 'A popup displays additional information on top of a page.',
  mixins: [_lib.SemanticUIVueMixin],
  props: {
    basic: {
      type: Boolean,
      description: 'Display the popup without the pointing arrow.'
    },
    content: {
      type: String,
      description: 'Simple text content for the popover.'
    },
    flowing: {
      type: Boolean,
      description: 'A flowing Popup has no maximum width and continues to flow to fit its content.'
    },
    header: {
      type: String,
      description: 'Header displayed above the content in bold.'
    },
    inverted: {
      type: Boolean,
      description: 'Invert the colors of the Popup.'
    },
    position: (0, _PropTypes.Enum)(_popupConstants.POSITIONS, {
      description: 'Position for the popover.',
      default: 'top left'
    }),
    size: (0, _PropTypes.Enum)(['mini', 'tiny', 'small', 'large', 'huge'], {
      description: 'Popup size.'
    }),
    wide: (0, _PropTypes.Enum)(['wide'], {
      type: Boolean,
      description: 'Popup width.'
    })
  },
  data: function data() {
    return {
      coords: null,
      open: false
    };
  },
  mounted: function mounted() {
    var _this = this;

    this.$slots.trigger[0].elm.addEventListener('mouseenter', this.handleOpen);
    this.$slots.trigger[0].elm.addEventListener('mouseleave', function () {
      _this.open = false;
    });
  },

  methods: {
    handleOpen: function handleOpen() {
      this.coords = this.$slots.trigger[0].elm.getBoundingClientRect();
      this.open = true;
    }
  },
  render: function render() {
    var h = arguments[0];

    return h(
      _VFragment2.default,
      null,
      [this.$slots.trigger, this.open && h(
        _PopupContainer2.default,
        {
          attrs: {
            popupClass: this.classes(this.basic && 'basic', this.flowing && 'flowing', this.inverted && 'inverted', this.size, this.wide, this.wide && 'wide'),
            triggerCoords: this.coords,
            position: this.position
          }
        },
        [this.header && h(
          _PopupHeader2.default,
          null,
          [this.header]
        ), this.content && h(
          _PopupContent2.default,
          null,
          [this.content]
        ), this.$slots.default]
      )]
    );
  }
};
